python_modules/dagster/dagster/_core/instance/__init__.py

```diff
@@ -1363,21 +1363,17 @@ def _log_materialization_planned_event_for_asset(
         from dagster._core.definitions.partition_key_range import PartitionKeyRange
         from dagster._core.events import AssetMaterializationPlannedData, DagsterEvent
 
+        is_partitioned = check.not_none(output.properties).is_asset_partitioned
         partition_tag = dagster_run.tags.get(PARTITION_NAME_TAG)
         partition_range_start, partition_range_end = (
             dagster_run.tags.get(ASSET_PARTITION_RANGE_START_TAG),
             dagster_run.tags.get(ASSET_PARTITION_RANGE_END_TAG),
         )
 
-        if partition_tag and (partition_range_start or partition_range_end):
-            raise DagsterInvariantViolationError(
-                f"Cannot have {ASSET_PARTITION_RANGE_START_TAG} or"
-                f" {ASSET_PARTITION_RANGE_END_TAG} set along with"
-                f" {PARTITION_NAME_TAG}"
-            )
```

Comment:
this restriction no longer makes sense since runs targeting a single partition but launched as part of a backfill will have both of these set